web/controllers/account_controller.ex

```diff
@@ -1,14 +1,42 @@
 defmodule Entice.Web.AccountController do
   use Entice.Web.Web, :controller
   alias Entice.Web.Account
+  alias Entice.Web.Invitation
+  alias Entice.Web.Queries
+
+  plug :ensure_login when action in [:request_invite]
 
   def create(conn, _params) do
     email = conn.params["email"]
     password = conn.params["password"]
+    invite_key = conn.params["inviteKey"]
 
-    %Account{email: email, password: password}
-    |> Entice.Web.Repo.insert
+    result = case Queries.get_invite(email) do    
+      {:error, :no_matching_invite} ->
+        error(%{message: "No Invitation found for this Email"})
+      {:ok, %Invitation{key: ^invite_key} = invite} ->
+        %Account{email: email, password: password}
+          |> Entice.Web.Repo.insert
+        # Delete the used invite (no need to store them)
+        Repo.delete(invite)
+        ok(%{message: "Account created!"})
+      {:ok, _} ->
```

Comment:
Tiny one statement cases like these can also go into one line, like in the second method below ;)